fabtests/component/sock_test.c

```diff
@@ -0,0 +1,253 @@
+#include <stdio.h>
+#include <string.h>  
+#include <stdlib.h>
+#include <errno.h>
+#include <unistd.h>   
+#include <arpa/inet.h>    
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <sys/time.h>
+#include <netdb.h>
+#include <time.h>
+#include <poll.h>
+#include <sys/epoll.h>
+
+#define PORT 8888
+
+size_t server_addr_len;
+
+int get_server_addr(char* server_addr, struct sockaddr_storage *server_sock, int *sock)
+{
+	struct addrinfo *res;
+	struct sockaddr_in *in;
+	struct sockaddr_in6 *in6;
+	int ret;
+	int optval = 1;
+
+	ret = getaddrinfo(server_addr, NULL, NULL, &res);
+	if (ret) {
+		printf("getaddrinfo failed\n");
+		return ret;
+	}
+
+	memcpy(server_sock, res->ai_addr, res->ai_addrlen);
+	server_addr_len = res->ai_addrlen;

```

Comment:
No need to copy the address.  Use res->ai_addr directly.  Also call freeaddrinfo(res) at the function end, including in any error case.